cmd/admin-handlers-users.go

```diff
@@ -516,7 +516,7 @@ func (a adminAPIHandlers) DeleteServiceAccount(w http.ResponseWriter, r *http.Re
 		return
 	}
 
-	if cred.AccessKey != user {
+	if cred.AccessKey != user && cred.ParentUser != user {
```

Comment:
Not sure what the second condition means ?

Besides, this is AND condition. It means that if cred.ParentUser is empty (like for all the users), this condition will never be true and any user can delete the service accounts of another user.

Reply:
It should be an OR @vadmeste